app/controllers/structures_controller.rb

```diff
@@ -18,13 +18,18 @@ def show
   def update
     authorize! :manage_item, @cocina
 
-    status = StructureUpdater.from_csv(@cocina, params[:csv].read)
-
-    if status.success?
-      object_client.update(params: @cocina.new(structural: status.value!))
-      redirect_to solr_document_path(@cocina.externalIdentifier), status: :see_other, notice: 'Structural metadata updated'
+    state_service = StateService.new(@cocina.externalIdentifier, version: @cocina.version)
+    if state_service.allows_modification?
+      status = StructureUpdater.from_csv(@cocina, params[:csv].read)
+
+      if status.success?
+        object_client.update(params: @cocina.new(structural: status.value!))
+        redirect_to solr_document_path(@cocina.externalIdentifier), status: :see_other, notice: 'Structural metadata updated'
+      else
+        redirect_to solr_document_path(@cocina.externalIdentifier), flash: { error: status.failure.join('\n') }
+      end
     else
-      redirect_to solr_document_path(@cocina.externalIdentifier), flash: { error: status.failure.join('\n') }
+      redirect_to solr_document_path(@cocina.externalIdentifier), flash: { error: 'Updates not allowed on this object.' }
     end
```

Comment:
Is this what you were thinking to ensure we don't update if locked?

Reply:
Yes, it just needs a test now.

Reply:
Added